runner.py

```diff
@@ -1099,7 +1099,11 @@ def stop_metric_providers(self):
             if stderr_read is not None and str(stderr_read):
                 errors.append(f"Stderr on {metric_provider.__class__.__name__} was NOT empty: {stderr_read}")
 
-            metric_provider.stop_profiling()
+            # pylint: disable=broad-exception-caught
+            try:
+                metric_provider.stop_profiling()
+            except Exception as exc:
```

Comment:
Why are you not catching more specific here? Couldn't you catch `ProcessLookupError` and `RuntimeError`?

Reply:
the reason for my choice here is that this is a sensitive function path and in any case you want to call the stop_profiling on every of the providers and not fail early due to an uncaught exception.